### ISSUE TITLE
Value/Index confusion in numbers example

### ISSUE THREAD

**Issue by rbu** [caje731/a_byte_of_python#1]
Hey Cajetan, thank you for the presentation at the Meetup which I enjoyed a lot.

I was a bit confused by this snippet:
```
x = [ a for a in range(-5,999)]
y = [ a for a in range(-5,999)]
z = []

for i in x:
    if x[i] == y[i] and x[i] is y[i]:
        z.append(x[i])
        
print(z)
```

While that code produces the expected result (listing all integers that are equal and also identical), the use of list *items* as *indices* seems like a bad practice to me. The only reason it works is that the list items are continuous and contain 0.

If you changed it as follows, it produces an IndexError:
```
x = [ a for a in range(-5,999)] + [1050]
y = [ a for a in range(-5,999)] + [1050]
```

I would propose a change to:
```
x = [ a for a in range(-5,999)]
y = [ a for a in range(-5,999)]
z = []

for ix, i in enumerate(x):
    if x[ix] == y[ix] and x[ix] is y[ix]:
        z.append(i)
        
print(z)
```

**Comment by caje731** [comment]
Wow, I never got notified that you raised this issue - I only accidentally stumbled on this.

You are indeed right, I may have overlooked this as part of last minute code changes before the talk. Thanks for bringing this to my attention!

**Comment by caje731** [comment]
Fixed with commit: 2980b4c94493370acfcedbc066eaa4e74e62c8a1
